examples/Migration/CampaignReportToCsv.cs

```diff
@@ -0,0 +1,118 @@
+        /// <param name="customerId">The Google Ads customer ID for which the call is made.</param>
+        public void Run(GoogleAdsClient client, long customerId,
+            string outputFilePath = OUTPUT_FILE_PATH)
+        {
+            GoogleAdsServiceClient googleAdsServiceClient =
+                client.GetService(Services.V8.GoogleAdsService);
+
+            // Create a query that retrieves campaigns.
+            string query = @"
+                SELECT
+                    campaign.id,
+                    campaign.name,
+                    segments.date,
+                    metrics.impressions,
+                    metrics.clicks,
+                    metrics.cost_micros 
+                FROM campaign 
+                WHERE segments.date DURING LAST_30_DAYS 
+                    AND campaign.status = 'ENABLED' 
+                ORDER BY segments.date DESC";
+
+            // Issues a search request.
+            googleAdsServiceClient.SearchStream(customerId.ToString(), query,
+                delegate(SearchGoogleAdsStreamResponse response)
+                {
+                    if (response.Results.Count() == 0)
+                    {
+                        Console.WriteLine("No results found!");
+                        return;
+                    }
+
+                    CsvFile csvFile = new CsvFile();
+
+                    // Set the header for the CSV file.
+                    csvFile.Headers.AddRange(response.FieldMask.Paths);
+
+                    // Iterate over all returned rows and extract the information.
+                    foreach (GoogleAdsRow googleAdsRow in response.Results)
+                    {
+                        csvFile.Records.Add(new string[]
+                        {
+                            googleAdsRow.Campaign.Id.ToString(),
+                            googleAdsRow.Campaign.Name,
+                            googleAdsRow.Segments.Date,
+                            googleAdsRow.Metrics.Impressions.ToString(),
+                            googleAdsRow.Metrics.Clicks.ToString(),
+                            googleAdsRow.Metrics.CostMicros.ToString()
+                        });
+                    }
+
+                    if (outputFilePath == null)
+                    {
+                        outputFilePath =
+                            Environment.GetFolderPath(Environment.SpecialFolder.UserProfile) +
+                            Path.DirectorySeparatorChar +
+                            GetType().Name +
+                            DateTime.Now.ToString("-yyyyMMMMdd-HHmmss") + ".csv";
+                    }
+                    else if (!outputFilePath.EndsWith(".csv"))
+                    {
+                        outputFilePath += ".csv";
+                    }
+
+                    // Create the file with the specified path, write all lines, and close it.
+                    // File.WriteAllText(outputFilePath, csvRows.ToString());
```

Comment:
Nit: you might want to delete this commented line.

Reply:
Right, my mistake; left it in while testing. Removed.